source/distributed/tria.cc

```diff
@@ -3361,6 +3361,20 @@ namespace parallel
               // get cell weights for a weighted repartitioning.
               const std::vector<unsigned int> cell_weights = get_cell_weights();
 
+#  ifdef DEBUG
+              // verify that the global sum of weights is larger than 0
+              const auto local_weights_sum =
+                std::accumulate(cell_weights.begin(),
+                                cell_weights.end(),
+                                std::uint64_t(0));
+              const auto global_weights_sum =
+                Utilities::MPI::sum(local_weights_sum, this->mpi_communicator);
+              Assert(global_weights_sum > 0,
```

Comment:
If you find that
```
Assert(Utilities::MPI::sum(std::accumulate(cell_weights.begin(),
                                           cell_weights.end(),
                                           std::uint64_t(0))) > 0,
       ExcMessage(
         "The global sum of weights over all active cells "
         "is zero. Please verify how you generate weights."));
```
is easier to read than I will change these parts accordingly.

I liked that the variable names describe what each part is actually representing. And it also makes the exception message more descriptive in my opinion. But I don't feel strongly about it.

Reply:
The presence of significant code blocks inside `#ifdef DEBUG ... #endif` always makes me a bit nervous because of the potential to actually do something only in debug but not release mode. 

But I don't feel strongly about it. I don't think that the assert as you write it is too difficult to read.